service/worker/scheduler/workflow_test.go

```diff
@@ -1586,6 +1586,11 @@ func (s *workflowSuite) TestUpdate() {
 							},
 							Action: s.defaultAction("newid"),
 						},
+						SearchAttributes: &commonpb.SearchAttributes{
+							IndexedFields: map[string]*commonpb.Payload{
+								"myfield": payload.EncodeString("another value"),
```

Comment:
test that this got upserted?

Reply:
I modified the functional tests to check. I don't know how to get the schedule workflow search attributes in this test framework.

Reply:
I think you can use OnUpsertSearchAttributes. ideally it would be good to test all the branches of that loop/conditional: if an attribute is present and unchanged, present and changed, present and deleted, not present and set.

Reply:
I cannot. Because of how the test is setup with `runAcrossContinue`, I would only be able to add a `OnUpsertSearchAttribute.Maybe`, which kind of defeats the purpose here because we won't know for sure if it ran or not.